lib/galaxy/managers/workflows.py

```diff
@@ -1059,13 +1079,19 @@ def _workflow_to_dict_run(self, trans, stored, workflow, history=None):
                 }
                 for oc in step.output_connections
             ]
+            annotations_dict = {
+                "annotation": stored.annotations[0].annotation,
+                "update_time": (stored.annotations[0].stored_workflow.update_time).isoformat(),
+                "tags": self.convert_to_dict_from_many(stored.annotations[0].stored_workflow.tags),
+            }
             if step.annotations:
                 step_model["annotation"] = step.annotations[0].annotation
             if step.upgrade_messages:
                 step_model["messages"] = step.upgrade_messages
             step_models.append(step_model)
         return {
             "id": trans.app.security.encode_id(stored.id),
+            "annotation": annotations_dict,
```

Comment:
You're including properties here that are not part of the annotation model.
This is a good attempt that goes in the right direction, I'll try to find some time to rework this.

Reply:
Sure - any insights & examples are appreciated too.